app/src/main/java/org/hackillinois/androidapp2019/view/ScannerFragment.kt

```diff
@@ -1,219 +1,145 @@
-                    val staffOverride = staffOverrideSwitch.isChecked
-                    val hasCheckedIn = true
-                    val hasPickedUpSwag = true
-                    val checkIn = CheckIn(userId, staffOverride, hasCheckedIn, hasPickedUpSwag)
-
-                    viewModel.checkInUser(checkIn)
+            view.codeScannerView.setOnClickListener { codeScanner.startPreview() }
 
+            // get camera permission
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (context.checkSelfPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+                    requestPermissions(arrayOf(Manifest.permission.CAMERA), PERMISSIONS_REQUEST_ACCESS_CAMERA)
                 } else {
-                    val userEventPair = UserEventPair(lastEventScannedFor, userId)
-                    viewModel.markUserAsAttendingEvent(userEventPair)
+                    codeScanner.startPreview()
                 }
+            } else {
+                codeScanner.startPreview()
             }
         }
+
+        return view
     }
 
-    /**
-     * A callback that is called when the status of the last scan request changes.
-     */
-    fun processLastScan(lastScanStatus: ScanStatus?) {
-
-        when (lastScanStatus?.lastScanWasSuccessful) {
-            false -> {
-                AlertDialog.Builder(context).apply {
-                    setMessage("User not registered, or already scanned in for event.")
-                    setPositiveButton("Ok") { dialog, _ ->
-                        shouldParseScan = true
-                        dialog.cancel()
-                    }
-                    create()
-                }.show()
-            }
-            true -> {
-                lastSuccessfullyScannedUser = lastScanStatus.userId
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
 
-                AlertDialog.Builder(context).apply {
-                    setMessage("Success: $lastSuccessfullyScannedUser")
-                    setPositiveButton("Ok") { dialog, _ ->
-                        shouldParseScan = true
-                        dialog.cancel()
-                    }
-                    create()
-                }.show()
-                staffOverrideSwitch.isChecked = false
-            }
-            null -> {
-                shouldParseScan = true
-                return
+        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                codeScanner.startPreview()
             }
         }
+    }
 
+    override fun onResume() {
+        super.onResume()
+        codeScanner.startPreview()
+    }
+
+    override fun onPause() {
+        codeScanner.releaseResources()
+        super.onPause()
```

Comment:
The super method should always be first, unless it makes logical sense to put it second here